pkg/cmd/server/origin/master.go

```diff
@@ -609,8 +609,8 @@ func (c *MasterConfig) RunBuildImageChangeTriggerController() {
 func (c *MasterConfig) RunDeploymentController() {
 	_, kclient := c.DeploymentControllerClients()
 	env := []api.EnvVar{
-		{Name: "KUBERNETES_MASTER", Value: c.MasterAddr},
-		{Name: "OPENSHIFT_MASTER", Value: c.MasterAddr},
+		{Name: "KUBERNETES_MASTER", Value: c.MasterPublicAddr},
```

Comment:
Thanks @liggitt I'll take a look at how the service is starting up.  If there is a use case where public address is not accessible in the cluster then this is certainly wrong